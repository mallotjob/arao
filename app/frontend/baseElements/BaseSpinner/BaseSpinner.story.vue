<template>
  <Story
    title="BaseElements / BaseSpinner"
    :layout="{type: 'grid', width: '100%'}"
  >
    <Variant title="Default state">
      <BaseSpinner font="slim" />
      <BaseSpinner color="danger" />
      <BaseSpinner font="fat" />
      <BaseSpinner
        font="fatter"
        color="success"
      />
    </Variant>

    <Variant title="Size small">
      <BaseSpinner
        font="slim"
        type="ringY"
        color="light"
      />
      <BaseSpinner type="ringY" />
      <BaseSpinner
        font="fat"
        type="ringY"
        color="dark"
      />
      <BaseSpinner
        font="fatter"
        type="ringY"
        color="warning"
      />
    </Variant>
  </Story>
</template>

<script setup>
import BaseSpinner from './BaseSpinner.vue';

</script>
<style lang="scss" scoped>


</style>
