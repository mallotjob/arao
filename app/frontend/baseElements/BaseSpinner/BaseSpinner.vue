<template>
  <div
    class="spinner"
    :class="{
      ['size-' + size]: size,
      ['color-' + color]: color,
      ['font-' + font]: font,
      ['type-' + type]: type,
    }"
  />
</template>

<script setup>
defineProps({
  color: {
    default: 'primary',
    type: String,
    validator: value => {
      return ['primary', 'success', 'danger', 'warning', 'light', 'dark'].includes(value);
    }
  },
  size: {
    default: 'base',
    type: String,
    validator: value => {
      return ['xs', 'sm', 'base', 'lg', 'xl'].includes(value);
    }
  },
  font: {
    default: 'base',
    type: String,
    validator: value => {
      return ['slim', 'base', 'fat', 'fatter'].includes(value);
    }
  },
  type: {
    default: 'base',
    type: String,
    validator: value => {
      return ['base', 'ringY'].includes(value);
    }
  },
});


</script>
<style lang="scss" scoped>
.spinner {
  @apply w-12 h-12 rounded-full animate-spin
  border border-solid shadow-md;

  &.size {

  }

  &.color {
    &-primary {
     @apply border-blue-700 border-t-transparent;
    }

    &-dark {
      @apply border-gray-800 border-t-transparent;
    }

    &-success {
      @apply border-green-600 border-t-transparent;
    }

    &-danger {
      @apply border-red-700 border-t-transparent;
    }

    &-light {
      @apply border-white hover:border-gray-100 border-t-transparent;
    }

    &-warning {
      @apply border-yellow-400 border-t-transparent;
    }
  }

  &.font {
    &-base {
      @apply border-2;
    }

    &-fat {
      @apply border-4;
    }

    &-fatter {
      @apply border-8;
    }
  }

  &.type {
    &-ringY {
      @apply border-0 border-y;

      &.font {
        &-base {
          @apply border-y-2;
        }

        &-fat {
          @apply border-y-4;
        }

        &-fatter {
          @apply border-y-8;
        }
      }
    }
  }

}

</style>
