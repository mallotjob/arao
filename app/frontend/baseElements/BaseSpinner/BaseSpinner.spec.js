import { mount } from '@vue/test-utils';

import BaseSpinner from './BaseSpinner.vue';

describe('BaseSpinner', () => {
  it('has a default size', () => {
    const wrapper = mount(BaseSpinner);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should has a small[sm] size', () => {
    const wrapper = mount(BaseSpinner, { propsData: { size: 'sm' } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
