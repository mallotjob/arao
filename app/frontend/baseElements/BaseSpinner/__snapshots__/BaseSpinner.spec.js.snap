// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseSpinner > has a default size 1`] = `"<span class="spinner-border" role="status" aria-hidden="true"></span>"`;

exports[`BaseSpinner > has a small[sm] size  1`] = `"<span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>"`;
